records_mover/db/loader.py

```diff
@@ -80,21 +82,32 @@ def load_from_fileobj(self, schema: str, table: str,
         """
         ...
 
-    # Implement load() in terms of load_from_fileobj()
+    def load_from_records_directory_via_fileobj(self,
+                                                schema: str,
+                                                table: str,
+                                                load_plan: RecordsLoadPlan,
+                                                directory: RecordsDirectory) -> Optional[int]:
+        all_urls = directory.manifest_entry_urls()
+
+        total_rows = None
+        for url in all_urls:
+            loc = self.url_resolver.file_url(url)
+            with loc.open() as f:
+                logger.info(f"Loading {url} into {schema}.{table}...")
+                out = self.load_from_fileobj(schema, table, load_plan, f)
+                if out is not None:
+                    if total_rows is None:
+                        total_rows = 0
+                    total_rows += out
+        return total_rows
+
     def load(self,
              schema: str,
              table: str,
              load_plan: RecordsLoadPlan,
              directory: RecordsDirectory) -> Optional[int]:
-        all_urls = directory.manifest_entry_urls()
-
-        locs = [self.url_resolver.file_url(url) for url in all_urls]
-        fileobjs: List[IO[bytes]] = []
-        with ExitStack() as stack:
-            fileobjs = [stack.enter_context(loc.open()) for loc in locs]
-            concatted_fileobj: IO[bytes] = ConcatFiles(fileobjs)  # type: ignore
-            return self.load_from_fileobj(schema,
-                                          table,
-                                          load_plan,
-                                          concatted_fileobj)
-        return None  # make mypy happy
+        # Implement load() in terms of load_from_fileobj() by default
+        return self.load_from_records_directory_via_fileobj(schema=schema,
+                                                            table=table,
+                                                            load_plan=load_plan,
+                                                            directory=directory)
```

Comment:
This logic only worked for files which could be concatenated - and didn't have any safety checks to avoid sending the database invalid data files.  Basically, it worked for CSV files without headers, or single files.

The new logic simply loads multiple streams in a row, and relies on load_from_fileobj() having append semantics (I documented that requirement and verified the existing implementations comply).